### PR TITLE
Microchip: XEC GPIO driver interrupt enable fix part 2

### PR DESCRIPTION
**Pull request by scottwcpg**
Fixes issue #34879
This PR updates previous PR's 37138 and 37139.
Refer to issue 34879 for information from MCHP HW
designers. A delay after enabling interrupts is a
more appropriate work-around than depending upon
behavior of ARM DMB instruction.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

### CODE REVIEW

**Review by albertofloyd — APPROVED** [review]

File: drivers/gpio/gpio_mchp_xec.c [hunk 1]
```diff
@@ -215,7 +219,13 @@ static int gpio_xec_pin_interrupt_configure(const struct device *dev,
 	 */
 	current_pcr1 = config->pcr1_base + pin;
 	*current_pcr1 = (*current_pcr1 & ~mask) | pcr1;
-	__DMB(); /* insure write completes */
+	/* delay for HW to synchronize after it ungates its clock */
+	for (int i = 0; i < XEC_GPIO_EDGE_DLY_COUNT; i++) {
+		mask = *current_pcr1;
+		mask <<= 1;
+	}
```
Comment (mnkp) [inline drivers/gpio/gpio_mchp_xec.c:226]: An easier way to write the loop would be
```
	for (int i = 0; i < XEC_GPIO_EDGE_DLY_COUNT; i++) {
		(void)*current_pcr1;
	}
```
In this case there is no need for `sys_write32(mask, XEC_GPIO_DLY_ADDR);` The `XEC_GPIO_DLY_ADDR` define can be removed.

`current_pcr1` is defined as pointer to volatile so compiler will not optimize it away.

File: drivers/gpio/gpio_mchp_xec.c [hunk 2]
```diff
@@ -9,13 +9,17 @@
...
 #include "gpio_utils.h"
 
-#define XEC_GPIO_EDGE_DLY_COUNT		8
+#define XEC_GPIO_EDGE_DLY_COUNT		4
 /* read only register in same AHB segmment for dummy writes */
-#define XEC_GPIO_DLY_ADDR		0x40080150u
+#define XEC_GPIO_DLY_ADDR		\
+	((DT_REG_ADDR(DT_NODELABEL(pcr))) + MCHP_PCR_CLK_REQ0_OFS)
+
+#define XEC_ECIA_REGS ((struct ecia_regs *)(DT_REG_ADDR(DT_NODELABEL(ecia))))
```
Comment (mnkp) [inline drivers/gpio/gpio_mchp_xec.c:22]: The define seems to be unused and not related to PR. Should be probably removed.

File: drivers/gpio/gpio_mchp_xec.c (hunk 1 above)
Comment (scottwcpg) [reply drivers/gpio/gpio_mchp_xec.c]: @mnkp Changes done

File: drivers/gpio/gpio_mchp_xec.c (hunk 2 above)
Comment (scottwcpg) [reply drivers/gpio/gpio_mchp_xec.c]: @mnkp Changed done

**Review by mnkp — APPROVED** [review]

**Pull request merged**